### PR TITLE
If container does not have tty, 'exec' and then connect

### LINKED ISSUE #297: Terminal should 'exec' shell and connect to it
When command is not interactive shell, then terminal only logs instead of being usable as terminal.

### PR DESCRIPTION
**Pull request by marusak**
Fixes #297

### CODE REVIEW

File: src/ContainerTerminal.jsx [hunk 1]
```diff
@@ -95,7 +99,7 @@ class ContainerTerminal extends React.Component {
         var realWidth = this.state.term._core._renderService.dimensions.actualCellWidth;
         var cols = Math.floor((width - padding) / realWidth);
         this.state.term.resize(cols, 24);
-        client.resizeContainersTTY(this.props.system, this.state.container, cols, 24)
+        client.resizeContainersTTY(this.props.system, this.state.sessionId, this.props.tty, cols, 24)
```
Comment (KKoukiou) [inline src/ContainerTerminal.jsx:102]: Why don't you keep using the existing state 'container' here? It has the same value. (Or does it change somewhere I don't see?)

Comment (marusak) [reply src/ContainerTerminal.jsx:102]: It does, when we connect through exec, it is not container ID, but exec ID. It is set up [here](https://github.com/cockpit-project/cockpit-podman/pull/461/files#diff-ce6eac7021a57afa8c484caf011f3e15R195)

Comment (KKoukiou) [reply src/ContainerTerminal.jsx:102]: Ack thanks, thought that's still the containerId but aparently it's different value of that (r.Id)

File: src/Containers.jsx [hunk 2]
```diff
@@ -132,7 +134,7 @@ class Containers extends React.Component {
         }, {
             name: _("Console"),
             renderer: ContainerTerminal,
-            data: { containerId: container.Id, containerStatus: container.State, width:this.state.width, system:container.isSystem }
+            data: { containerId: container.Id, containerStatus: container.State, width:this.state.width, system:container.isSystem, tty: tty }
```
Comment (KKoukiou) [inline src/Containers.jsx:137]: not big of a deal - but this is a new property here, maybe you want to add it in PropTypes. (which is missing  half of the others already so we can fix this in the future)

Comment (marusak) [reply src/Containers.jsx:137]: ah right, I added the new one and also added the forgotten ones. Thanks for spotting!

**Pull request merged**